Sources/AsyncServer.swift

```diff
@@ -1,9 +1,10 @@
 public protocol AsyncServer {
-    func serve(responder: AsyncResponder, on host: String, at port: Int) throws
+    init(host: String, port: Int, responder: AsyncResponder) throws
+    func start() throws
 }
 
 extension AsyncServer {
-    public func serve(responder: AsyncResponder, at port: Int) throws {
-        try self.serve(responder, on: "0.0.0.0", at: port)
+    public init(port: Int, responder: AsyncResponder) throws {
```

Comment:
While I'm not a fan of protocols that force initializers it does make sense.

LGTM